tests/modules/images/webp-uploads/helper-tests.php

```diff
@@ -454,4 +454,57 @@ public function it_should_never_discard_additional_image_if_filter_is_false( $or
 		$output = webp_uploads_should_discard_additional_image_file( $original_filesize, $additional_filesize );
 		$this->assertFalse( $output );
 	}
+
+	public function test_webp_uploads_in_frontend_body_without_wp_query() {
+		unset( $GLOBALS['wp_query'] );
+
+		$this->assertFalse( webp_uploads_in_frontend_body() );
+	}
+
+	public function test_webp_uploads_in_frontend_body_with_feed() {
+		$this->mock_empty_action( 'template_redirect' );
+		$GLOBALS['wp_query']->is_feed = true;
+
+		$this->assertFalse( webp_uploads_in_frontend_body() );
+	}
+
+	public function test_webp_uploads_in_frontend_body_without_template_redirect() {
+		$this->assertFalse( webp_uploads_in_frontend_body() );
+	}
+
+	public function test_webp_uploads_in_frontend_body_before_template_redirect() {
+		$result = webp_uploads_in_frontend_body();
+		$this->mock_empty_action( 'template_redirect' );
+
+		$this->assertFalse( $result );
+	}
+
+	public function test_webp_uploads_in_frontend_body_after_template_redirect() {
+		$this->mock_empty_action( 'template_redirect' );
+		$result = webp_uploads_in_frontend_body();
+
+		$this->assertTrue( $result );
+	}
+
+	public function test_webp_uploads_in_frontend_body_within_wp_head() {
```

Comment:
More of nit-pick here but the test function names are different than what we actually use the names here like `it_should` or `it_` like that. And these functions doesn't have function doc comment.

cc: @felixarntz @akshitsethi

Reply:
@mehulkaklotar I didn't think it was too critical to point it out in earlier tests for the plugin, but the convention for WordPress core tests is to always use `test_` followed by the function name, then optionally anything else. So I stuck with that here.

I would advise to follow that convention for future tests to monitor the core plan.

Regarding doc comments, test methods do not require any doc comments per core specifications - only when helpful. Preferably, the name of the method should be clear enough to know what is going on, or when it makes sense to clarify specific code, inline comments should be used.

cc @akshitsethi @adamsilverstein @mukeshpanchal27 @jjgrainger @eugene-manuilov